api_tests/nodes/views/test_node_list.py

```diff
@@ -346,6 +376,31 @@ def test_incorrect_filtering_field_not_logged_in(self):
         assert_equal(len(errors), 1)
         assert_equal(errors[0]['detail'], "'notafield' is not a valid field for this endpoint.")
 
+    def test_filtering_on_root(self):
+        # Build up a family of nodes
+        node_structure = [5, 2, [1, 2], 3, [10, 2], 2, [4, [2, 2], 6]]
+        render_generations_from_node_structure_list(self.project_one, node_structure)
```

Comment:
I'm a fan of awesomethoroughnomics so long as the 'awesome' is simple to understand. If we need to test deeply nested relationships, you can create those easily:

```python
root = ProjectFactory()
child = ProjectFactory(parent=root)
sibling = ProjectFactory(parent=root)
grandchild = ProjectFactory(parent=child)
```

Now, if something breaks in the test, you can throw a breakpoint in and inspect the above fixtures if need be.